uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeavePeriodEntitlementTest.php

```diff
@@ -443,7 +443,7 @@ public function testCanSaveALeavePeriodEntitlementFromAnEntitlementCalculation()
       return $balanceChange->type_id == $balanceChangeTypes['leave'];
     });
     $this->assertCount(1, $leaveBalanceChanges);
-    $this->assertEquals($proRata, reset($leaveBalanceChanges)->amount);
+    $this->assertEquals($proRata - $numberOfPublicHolidays, reset($leaveBalanceChanges)->amount);
```

Comment:
Why do we have to subtract the number of public holidays from proRata?

Reply:
Because the number of public holidays is included in the calculation pro rata, as it can be seen [here](https://github.com/civicrm/civihr/blob/f2120b399208879e80912e70cee612b6db4c71bf/uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/ContractEntitlementCalculation.php#L40-L61)

However, we need two separate Leave Balance Changes. One only with the number of public holidays and another one only with the "Leave" amount. So [here](https://github.com/civicrm/civihr/blob/c34632d99947ad7390c135e4a95fea38defcb641/uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeavePeriodEntitlement.php#L236), we separate the values when the balance changes are created